tests/test_helper/microovn.bash

```diff
@@ -102,3 +119,31 @@ function microovn_get_cluster_address() {
     lxc_exec "$container" "microovn status" | \
         awk -F\( "/$container/{sub(/\)\$/,\"\");print\$2}"
 }
+
+# wait_for_open_port CONTAINER PORT MAX_RETRY
+#
+# Return after specified PORT is open and listening in the CONTAINER.
+#
+# The port can be bound to any interface and it's a good way to test whether
+# OVN services are up and listening. This function will retry for maximum of
+# MAX_RETRY times, each time backing of for 1 second between attempts.
+function wait_for_open_port() {
+    local container=$1
+    local port=$2
+    local max_retry=$3
+    local attempt=1
+    local success=0
+
+    while ! lxc_exec "$container" "lsof -i:$port -sTCP:LISTEN"; do
+        echo "# ($container) waiting for port $port to be opened ($attempt/$max_retry)"
+        if [ $attempt -gt "$max_retry" ]; then
+            echo "# ($container) Maximum retries reached"
+            success=1
+            break
+        fi
+        ((++attempt))
+        sleep 1
```

Comment:
One second is a long time, could we use a shorter interval, at least for the first iteration?

Reply:
I was thinking about doing something like exponential backoff but that could add unnecessary complexity. My reasoning for this simple approach was:

- It's dead simple
- Number of retries **roughly** corresponds to the overall timeout in seconds
- `sleep` is invoked only after first failed attempt so reasonably fast machines won't experience it at all
  - This whole check was added only because GH CI was not fast enough with bringing services UP. I was not able to reproduce the error in my local environment. My machine is no beast and I pass this check on first attempt every time, meaning that the 1s sleep is never executed.

If you still think that the 1s is excessive, I'm open to add more granular solution.